xdsl/dialects/arith.py

```diff
@@ -81,6 +81,13 @@ def from_int_constant(val: Union[int, Attribute],
             result_types=[typ],
             attributes={"value": IntegerAttr.from_params(val, typ)})
 
+    @staticmethod
+    def from_float32_constant(val: Union[float, Attribute],
+                            typ: Float32Type) -> Constant:
```

Comment:
You already know the type (its the Float32Type), so why do you still pass it as an argument? The difference to the int_constant constructor here is that it can take different bitwidths, whereas your constructor seems to be aimed at only 32-bit floats.